pkg/cmd/server/start/start_node.go

```diff
@@ -189,24 +202,7 @@ func (o NodeOptions) StartNode() error {
 // 3.  Writes the fully specified node config and exits if needed
 // 4.  Starts the node based on the fully specified config
 func (o NodeOptions) RunNode() error {
-	if !o.IsRunFromConfig() || o.IsWriteConfigOnly() {
-		glog.V(2).Infof("Generating node configuration")
-		if err := o.CreateNodeConfig(); err != nil {
-			return err
-		}
-	}
-
-	if o.IsWriteConfigOnly() {
-		return nil
-	}
-
-	var nodeConfig *configapi.NodeConfig
-	var err error
-	if o.IsRunFromConfig() {
-		nodeConfig, err = configapilatest.ReadAndResolveNodeConfig(o.ConfigFile)
-	} else {
-		nodeConfig, err = o.NodeArgs.BuildSerializeableNodeConfig()
-	}
+	nodeConfig, err := o.resolveNodeConfig()
```

Comment:
return the resolved config file so you don't have to modify `o` inside `resolveNodeConfig`